File: app/routers/notifications.py
from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session
from app.database import get_db
from app.routers.auth import get_current_user
from app.models.notification import Notification
from app.schemas import NotificationResponse

router = APIRouter(prefix="/notifications", tags=["notifications"])

@router.get("", response_model=list[NotificationResponse])
def get_notifications(
    db: Session = Depends(get_db),
    current_user=Depends(get_current_user),
):
    return db.query(Notification).filter(
        Notification.user_id == current_user.id
    ).order_by(Notification.created_at.desc()).limit(50).all()

@router.post("/{notification_id}/read")
def mark_read(
    notification_id: int,
    db: Session = Depends(get_db),
    current_user=Depends(get_current_user),
):
    n = db.query(Notification).filter(
        Notification.id == notification_id,
        Notification.user_id == current_user.id
    ).first()
    if n:
        n.read = 1
        db.commit()
    return {"ok": True}

@router.post("/read-all")
def mark_all_read(
    db: Session = Depends(get_db),
    current_user=Depends(get_current_user),
):
    db.query(Notification).filter(
        Notification.user_id == current_user.id,
        Notification.read == 0
    ).update({"read": 1})
    db.commit()
    return {"ok": True}

File: frontend/src/api/notifications.ts
import api from "./client";

export type Notification = {
  id: number;
  user_id: number;
  title: string;
  message: string;
  read: number;
  created_at: string;
};

export async function getNotifications() {
  const res = await api.get<Notification[]>("/notifications");
  return res.data;
}

export async function markRead(id: number) {
  await api.post("/notifications/" + id + "/read");
}

export async function markAllRead() {
  await api.post("/notifications/read-all");
}
